### PR TITLE
Fix warnings visible on CI builds

### PR DESCRIPTION
**Pull request by bradking**
Perform minor cleanup and refactoring to fix warnings that have appeared during CI (but not blocked merging) for a while.

### CODE REVIEW

**Comment by bradking** [comment]
The goal of this PR is to clear up warnings during CI, so please do not merge it if any remain.

**Comment by chuckatkins** [comment]
With this in place the I believe we can try to add back in the "warnings as errors" flags in the appropriate `scripts/ci/cmake/` ctests scripts that it's been disabled in.

**Comment by bradking** [comment]
> try to add back in the "warnings as errors" flags 

I added a commit to do that.

File: scripts/ci/cmake/ci-suse-pgi.cmake [hunk 1]
```diff
@@ -8,9 +8,9 @@ env_module(load pgi)
 set(ENV{CC}  pgcc)
 set(ENV{CXX} pgc++)
 set(ENV{FC}  pgfortran)
-#set(ENV{CFLAGS} -Werror)
+set(ENV{CFLAGS} -Werror)
 set(ENV{CXXFLAGS} --brief_diagnostics)
```
Comment (bradking) [inline scripts/ci/cmake/ci-suse-pgi.cmake]: @chuckatkins the PGI C compiler does not understand `-Werror`.  Do you know the correct options to use here for C and C++?

Comment (chuckatkins) [reply scripts/ci/cmake/ci-suse-pgi.cmake]: AFAIK it doesn't have such an option.

**Comment by bradking** [comment]
Yay, no warnings and `/WX` is now enforced on MSVC!

**Pull request merged**